src/Api/Orbit.cpp

```diff
@@ -8,74 +8,70 @@
 #include <absl/base/casts.h>
 
 #include "Api/EncodedEvent.h"
+#include "Api/Event.h"
 #include "Api/LockFreeApiEventProducer.h"
 #include "OrbitBase/Profiling.h"
 #include "OrbitBase/ThreadUtils.h"
 
-static void EnqueueApiEvent(orbit_api::EventType type, const char* name = nullptr,
-                            uint64_t data = 0, orbit_api_color color = kOrbitColorAuto) {
-  static orbit_api::LockFreeApiEventProducer producer;
-  if (!producer.IsCapturing()) return;
+namespace {
+std::unique_ptr<orbit_api::LockFreeApiEventProducer> producer;
+
+template <typename Event, typename... Types>
+void EnqueueApiEvent(Types... args) {
+  if (producer == nullptr) {
+    producer = std::make_unique<orbit_api::LockFreeApiEventProducer>();
+  }
```

Comment:
There is a risk of multiple initializations of "producer" from multiple threads here. I'd suggest wrapping it inside a function and using a static variable for thread safety, something like 

orbit_api::LockFreeApiEventProducer& GetEventProducer() {
static orbit_api::LockFreeApiEventProducer producer; 
return producer;
}

Reply:
Thats a good idea. I was wondering what would be the best to work around the prev. static producer in the now templated methods. Done.